### PR TITLE
Reduce container size

### PR DESCRIPTION
**Pull request by serverwentdown**
Reduces size by half, not sure why... I'll document why after some research.

https://hub.docker.com/r/serverwentdown/concourse-ansible-playbook-resource/tags

### CODE REVIEW

**Comment by serverwentdown** [comment]
Previously, PIP cache and cargo registry were taking up space in the `/root` folder.  

image

**Comment by troykinsella** [comment]
Thanks for looking into the image size!

I've always written Dockerfiles with sensitivity to minimizing the filesystem layers generated (by combining multiple commands into a single `RUN` directive, for example) for runtime performance reasons. From that angle, I hesitate to want to merge this PR.

That said, I've only adopted that mindset from best-practice suggestions I've seen out there. I haven't ever done any benchmarking myself. Maybe the performance impact is negligible these days.

**Comment by serverwentdown** [comment]
Oh! I never thought of build time being a potential concern, and since there's no reduction in readability by reducing layers, I'll make the changes.

**Comment by serverwentdown** [comment]
- [x] Build locally with Podman
- [x] Build on Docker Hub with Docker

**Comment by troykinsella** [comment]
Looking good so far!

I'm less-so concerned with build-time performance as I am with runtime performance (and again, maybe there's no need for concern there). I think at runtime, the time to access a file increases for every file system layer that needs to be traversed to get to it. And being that directives in a Dockerfile often generate a file system layer, I try to keep them to a minimum.

I also tend to keep the `set -eux` (the `-x` at least) at the beginning of a `RUN` directive because otherwise (at least as of last time I checked) Docker leaves you in the dark about which of the several commands failed if one does.

**Comment by serverwentdown** [comment]
Sorry, I misread your previous comment. `COPY` directives take up layers in the final stage. I'll try a different approach and update this PR.

Following the Dockerfiles of the official images, they seem to prefer using `&&` instead of `set -x`, that's why I'm using them here.

**Comment by troykinsella** [comment]
Ah no worries. I think `set -e` would substitute for `&&`, but `set -x` echos commands before running them. That's the part I'd want to keep around.

File: Dockerfile [hunk 1]
```diff
@@ -1,36 +1,76 @@
...
+RUN ansible-playbook --version
+
+# Install mockleton
+RUN apk add --update --no-cache bash
+RUN gem install rspec
+RUN wget -q -O - https://raw.githubusercontent.com/troykinsella/mockleton/master/install.sh | bash
+# Mock binaries used in testing
+RUN cp /usr/local/bin/mockleton /usr/bin/ansible-galaxy
+RUN cp /usr/local/bin/mockleton /usr/bin/ansible-playbook
+RUN cp /usr/local/bin/mockleton /usr/bin/ssh-add
```
Comment (gunziptarball) [inline Dockerfile]: Would it not be better to consolidate these commands into a single `RUN`?  Or is it better to keep them separate with this being a testing stage?  I'm somewhat new to multi-stage builds and don't use them regularly so forgive me if I don't know what I'm talking about 😄 

```suggestion
RUN echo 'Sanity: ensure we can run Ansible' \
  && ansible-playbook --version \
  && echo 'Install mockleton' \
  && apk add --update --no-cache bash \
  && gem install rspec \
  && wget -q -O - https://raw.githubusercontent.com/troykinsella/mockleton/master/install.sh | bash \
  && echo 'Mock binaries used in testing' \
  && cp /usr/local/bin/mockleton /usr/bin/ansible-galaxy \
  && cp /usr/local/bin/mockleton /usr/bin/ansible-playbook \
  && cp /usr/local/bin/mockleton /usr/bin/ssh-add
```

**Comment by troykinsella** [comment]
Looks good. Thanks!

**Pull request merged**